protocol-units/da-sequencer/node/src/batch.rs

```diff
@@ -100,91 +66,117 @@ pub fn deserialize_full_node_batch(
-        signature: &Signature,
-        public_key: &VerifyingKey,
+	batch_data: &[u8],
+	signature: &Signature,
+	public_key: &VerifyingKey,
 ) -> Result<(), DaSequencerError> {
-        public_key
-                .verify(batch_data, signature)
-                .map_err(|_| DaSequencerError::InvalidSignature)
+	public_key
+		.verify(batch_data, signature)
+		.map_err(|_| DaSequencerError::InvalidSignature)
 }
 
 #[cfg(test)]
 mod tests {
-        use super::*;
-        use movement_da_sequencer_client::sign_batch;
-        use movement_da_sequencer_config::DaSequencerConfig;
-        use tracing_subscriber;
-
-        #[test]
-        fn test_sign_and_validate_batch() {
-                let _ = tracing_subscriber::fmt()
-                        .with_max_level(tracing::Level::INFO)
-                        .with_test_writer()
-                        .try_init();
-        
-                let config = DaSequencerConfig::default();
-                let signing_key = config.signing_key;
-                let verifying_key = signing_key.verifying_key();
-        
-                // Create transactions and batch
-                let txs = FullNodeTxs(vec![
-                        Transaction::new(b"hello".to_vec(), 0, 1),
-                        Transaction::new(b"world".to_vec(), 0, 2),
-                ]);
-        
-                let batch_bytes = bcs::to_bytes(&txs).expect("Serialization failed");
-                let signature = sign_batch(&batch_bytes, &signing_key);
-        
-                // Serialize full node batch into raw bytes
-                let serialized = serialize_full_node_batch(
-                        verifying_key,
-                        signature.clone(),
-                        batch_bytes.clone(),
-                );
-        
-                // Deserialize it back
-                let (deserialized_key, deserialized_sig, deserialized_data) =
-                        deserialize_full_node_batch(serialized).expect("Deserialization failed");
-        
-                // Recreate the raw batch from deserialized data
-                let raw_batch = DaBatch {
-                        data: RawData { data: deserialized_data },
-                        signature: deserialized_sig,
-                        signer: deserialized_key,
-                        timestamp: chrono::Utc::now().timestamp_micros() as u64,
-                };
-        
-                // Validate the batch
-                let validated = validate_batch(raw_batch).expect("Batch should validate");
-        
-                // Check it worked
-                assert_eq!(validated.data.0.len(), 2);
-                assert_eq!(validated.data.0, txs.0);
-        }
-}
\ No newline at end of file
+	use super::*;
+	use aptos_crypto::hash::CryptoHash;
+	use ed25519_dalek::Signer;
+
+	use movement_da_sequencer_client::sign_batch;
+	use movement_da_sequencer_config::DaSequencerConfig;
+	use tracing_subscriber;
+
+	impl<D> DaBatch<D>
```

Comment:
I see you moved the impl block for this test method into the mod tests { } 

;) 

That's ok then, let's do it like this. It just means that this method cannot be used by other testing modules, but, perhaps it shouldn't anyway.

Reply:
We can export as public the mock part. I add it so you can use the mock in unit test.

Reply:
Done

Reply:
If it's done, just feel free to resolve the issue yourself 👍